R/utilities.R

```diff
@@ -140,3 +138,45 @@ safe_grobHeight <- function(grob, unitTo = "bigpts", valueOnly = TRUE){
 
   return(grid::convertHeight(grid::grobHeight(grob), unitTo, valueOnly))
 }
+
+
+#' Palette Fetcher
+#'
+#' @param which a vector of palette types to consider
+#' @param return Value to return. "colors", the default, returns the palette as
+#'   a vector of colors. "type" returns the palette's type. "Exists" returns
+#'   TRUE or FALSE based on whether the name is found in the palettes table.
+#'
+#' @describeIn viz_palette Return details about a palette
+#'
+#' @examples
+#' # Identify the first two colors of the Prosperity Palette
+#' fetch_pal("prosperity")[1:2]
+#'
+#' # Confirm that "reds" is a sequential palette
+#' fetch_pal("reds", which = "sequential", return = "exists")
+#'
+#' @export
+fetch_pal <- function(pal,
+                      which = c("discrete", "sequential", "divergent"), #unique(cmapplot_globals$palettes$type),
+                      return = c("colors", "type", "exists")){
+  # basics
+  name <- type <- NULL
+  return <- match.arg(return)
```

Comment:
could add, if we wanted: `which <- match.arg(which, unique(cmapplot_globals$palettes$type), several.ok = TRUE)`

Reply:
Would this display the allowed options in the help files?

Reply:
This is separate, but a response to your tweak. I'm suggesting we leave your change to the function defaults, but add a `match.arg` to confirm what the user calls is acceptable.